backend/src/services/payments.js

```diff
@@ -0,0 +1,63 @@
+const axios = require('axios');
+const crypto = require('crypto');
+const db = require('../db');
+
+const tokenUrl = 'https://payform.bambora.com/pbwapi/auth_payment';
+const bamboraApiUrl = 'https://payform.bambora.com/pbwapi/token/';
+
+const secret = process.env.SECRET_KEY;
+const apiKey = process.env.API_KEY;
+
+const payment_return_url = process.env.PAYMENT_RETURN_URL;
+const payemnt_notify_url = process.env.PAYMENT_NOTIFY_URL;
+
+const getPaymentRedirectUrl = async (paymentObj) => {
+    const orderNumber = 'vantaa-order-' + Date.now();
+    const message = apiKey + '|' + orderNumber;
+    const authCode = crypto
+        .createHmac('sha256', secret)
+        .update(message)
+        .digest('hex')
+        .toUpperCase();
+    paymentObj.auth_code = authCode;
+    paymentObj.order_number = orderNumber;
+    await db.payments.createPayment(paymentObj);
+    const response = await axios
```

Comment:
Would be good in the future to wrap all of the Bambora interaction to its own service as it tends to get a little verbose.